[AIRFLOW-1383] Add no_dag_reset option to airflow clear command

chronitis opened the pull request:
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!


### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
    - https://issues.apache.org/jira/browse/AIRFLOW-1383


### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

The CLI command `airflow clear` does not expose any way to set the `reset_dag_runs` parameter of the underlying `DAG.clear()` method.

This means that `airflow clear` will always cause affected DAG runs to be set to `state=RUNNING`, and immediately try and re-run the cleared tasks. In some circumstances it might not be desirable to immediately re-run everything.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

A simple test for the CLI syntax is added to `tests/core.py`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

Commit:
[AIRFLOW-1383] Add no_dag_reset option to airflow clear command

The underlying DAG.clear() function has a parameter reset_dag_runs
(default True) which controlls whether DAG runs for which a task was
cleared will be set to state=RUNNING. The current CLI doesn't expose
this option; this adds a flag (no_dag_reset) which prevents runs being
set to RUNNING after clearing, for cases where it is desirable to clear
jobs but not immediately have them re-running.

Commit:
[AIRFLOW-1383] Add a simple test case for --no_dag_reset

Commit:
[AIRFLOW-1383] Add Lifesum to airflow users list

ashb commented on airflow/bin/cli.py:
```diff
@@ -1339,6 +1340,9 @@ class CLIFactory(object):
             "Search dag_id as regex instead of exact string", "store_true"),
+        'no_dag_reset': Arg(
+            ("--no_dag_reset", ),
+            "Do not reset DAG state after clearing tasks", "store_true"),
```
Rather than having to do `reset_dag_runs=not args.no_dag_reset` at the call site, would we do something like `"store_false", metavar="reset_dag_runs", default=True)`?

ashb replied:
metavar was not what I meant.

`'reset_dag_runs': Arg(... "store_false", default=True)` was what I meant.

chronitis replied:
Not possible without further changes; the `Arg()` namedtuple doesn't include `dest`, and the generated variable name is not set by the dictionary key; you'd have to do `--reset_dag_runs=0` or something similar, which seems unnecessarily ugly.

ashb commented on tests/core.py:
```diff
@@ -1276,6 +1276,11 @@ def test_get_dags(self):
+    def test_clear_no_reset(self):
+        args = self.parser.parse_args([
+            'clear', 'example_python_operator', '--no_dag_reset', '--no_confirm'])
+        cli.clear(args)
```
I feel that some assertion that the flag does what we expect is needed here please.

(Late review, sorry!)

Commit:
[AIRFLOW-1383] Test constant DAG state

With --no_dag_reset, the state of the DAG should not change to RUNNING
after calling clear.
